### ISSUE TITLE
Incorrect Screenshot Resolution when Opening a New Tab in Chrome v128 (Taiko 1.4.1 release)

### ISSUE THREAD

**Issue by winstoniuss** [getgauge/taiko#2733]
**Describe the bug**

I often use _await screenshot()_  (it is necessary for my mechanism that compares screenshots using the PixelMatch).
In Chrome v.128, there is an issue: 
    if screenshots are taken in the main tab with the resolution specified in the setViewPort, then when opening a new tab, screenshots are taken with the browser display resolution. This issue occurs regardless of whether the headless parameter is set to true or false.

In Chrome v.126 and earlyer, this issue was not found.

**To Reproduce**
1. await openBrowser({ 
headless: false,
        args: [ "--window-size=1600,900",  "--disable-gpu", "--disable-renderer-backgrounding",
            "--disable-extensions", "--no-first-run", "--no-sandbox", "--disable-background-networking",
            "--disable-sync", "--disable-popup-blocking"] });

2. await setViewPort({ width: 1600, height: 900 });
3. await goto('https://the-internet.herokuapp.com/tables');
4. await screenshot(); 
5. await openTab('https://the-internet.herokuapp.com/checkboxes');
6. await screenshot(); 
7. await closeBrowser();

**Expected behavior**
I expect that screenshots will be taken with the resolution specified in the viewPort, regardless of which tab they are taken in. 

**Screenshots**
Screenshots will be attached here to visually demonstrate the issue.
![Screenshot-1721200278682](https://github.com/user-attachments/assets/6120f7d1-45ba-4f4d-8b54-6307d13c117d)
![Screenshot-1721200279865](https://github.com/user-attachments/assets/67694768-6189-495f-bb94-7de3790da583)

**Versions:**
 - Taiko: 1.4.1
 - OS Windows10
 - Node.js v18.18.2

**Comment by winstoniuss** [comment]
UPD: The problem is not that screenshots are taken in a different resolution, but that the resolution of the new tab window becomes different from the main one

**Comment by winstoniuss** [comment]
await openBrowser({
headless: false,
args: [ "--window-size=1600,900"] });
await setViewPort({ width: 1600, height: 900 });
await goto('https://the-internet.herokuapp.com/windows');
await screenshot();
await click(''Click Here)
await screenshot();
await closeBrowser();


It seems this issue is inherent to all "Chrome for testing" versions.
check it

**Comment by zabil** [comment]
This might be a windows only issue I don't this locally on the mac machines for example

```
const { openBrowser, setViewPort, goto, screenshot, click, closeBrowser } =
    require('taiko');

(async () => {
    try {
        await openBrowser({
            headless: false, args: ["--window-size=1600,900"]
        });
        await setViewPort({ width: 800, height: 600 });
        await goto('https://the-internet.herokuapp.com/windows');
        await screenshot();
        await click('Click Here');
        await screenshot();
    } catch (error) {
        console.error(error);
    } finally {
        await closeBrowser();
    }
})();
```

```
$ sips -g pixelWidth -g pixelHeight Screenshot-1721641283456.png                                                                                      10:43:24
/Users/zabilcm/projects/earthinitiative/Screenshot-1721641283456.png
  pixelWidth: 800
  pixelHeight: 600
```

Notice it takes the viewport size.

**Comment by winstoniuss** [comment]
step("test", async function () {
    await openBrowser({ headless: true, args: ["--window-size=1600,900"] });
    await setViewPort({ width: 800, height: 600 });
    await goto("https://the-internet.herokuapp.com/windows");
    await screenshot(); // screenshot resolution = 800x600
    await click("Click Here");
    await screenshot(); // screenshot resolution = 3074x1404
});


It might be a problem only on Windows and only on the latest versions of Chrome, but I am looking for a solution to this issue because I am testing a product that is primarily deployed and run on Windows.
If you have any thoughts on solving this problem, I would be glad to hear them and am willing to cooperate to resolve this issue.

**Comment by zabil** [comment]
In the sample you posted 

```
    await screenshot(); // screenshot resolution = 800x600
    await click("Click Here");
    await screenshot(); // screenshot resolution = 3074x1404
```

The `Click here` opens a new window. Which I suspect resets the viewport size. Can check the resolution by modifying the script to

```
step("test", async function () {
    await openBrowser({ headless: true, args: ["--window-size=1600,900"] });
    await setViewPort({ width: 800, height: 600 });
    await goto("https://the-internet.herokuapp.com/windows");
    await screenshot(); // screenshot resolution = 800x600
    await click("Click Here");
    await setViewPort({ width: 800, height: 600 });
    await screenshot(); 
});
```

**Comment by winstoniuss** [comment]
Yes, that's correct. When opening a new window (or a new tab), the resolution resets.
If you run your script (with the repeated setViewPort), there is no problem.

I am currently using Taiko + Gauge with a separate version of Chromium (v. 120.xxxx) and there is no such problem.

I want to switch to the current version of Taiko with "Chrome for testing v.128.xxxx", but the volume of tests is too large to redo all the tests in this way.

**Comment by zabil** [comment]
We'll need to look into this. But it will probably only be fixed if Chrome for Testing releases a new version with a fix.

**Comment by winst0niuss** [comment]
In Chrome for Testing v.133, the issue persists, and the viewport size is not passed to the new tab. It is reproducible on both Windows and Ubuntu. 
Is it possible to add a 'hack' as listening for the new tab opening event, and upon opening, immediately pass it a viewport-size?
